Mathlib/RingTheory/KrullDimension.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fangming Li, Jujian Zhang
+-/
+import Mathlib.AlgebraicGeometry.PrimeSpectrum.Basic
+import Mathlib.Topology.KrullDimension
+
+/-!
+# Krull dimensions of (commutative) rings
+
+Given a commutative ring, its ring theoretic Krull dimension is the order theoretic Krull dimension
+applied to its prime spectrum. Unfolding this definition, it is the length of the longest series of
+prime ideals ordered by inclusion.
+
+
+## Main definitions and results
+
+* `ringKrullDim`: the ring theoretic Krull dimension of a commutative ring.
+
+* `ringKrullDim_eq_topologicalKrullDim`: the ring theoretic Krull dimension of a commutative ring
+is equal to the topological Krull dimension of its prime spectrum.
+
+* `ringKrullDim_eq_zero_of_field`: the Krull dimension of a field is zero.
+-/
+
+/--
+The ring theoretic Krull dimension is the Krull dimension of its spectrum ordered by inclusion.
+-/
+noncomputable def ringKrullDim (R : Type*) [CommRing R] : WithBot (WithTop ℕ) :=
+  krullDim (PrimeSpectrum R)
+
+variable {R S : Type*} [CommRing R] [CommRing S]
+
+@[nontriviality]
+lemma ringKrullDim_eq_bot_of_subsingleton [Subsingleton R] :
+    ringKrullDim R = ⊥ :=
+  krullDim_eq_bot_of_isEmpty
+
+lemma ringKrullDim_nonneg_of_nontrivial [Nontrivial R] :
+    0 ≤ ringKrullDim R :=
+  krullDim_nonneg_of_nonempty
```

Comment:
We would probably want iff versions of these or else there is little reason to let `ringKrullDim` be valued in `WithBot`. We are missing the analogous lemmas for `krullDim` though so maybe we can add this in a separate PR.

Reply:
OK! I'll prove this in a new PR!